### PR TITLE
Change VerrazzanoProject to contain a list of namespace resources

### PR DESCRIPTION
**Pull request by mgianatagh**
# Description

Change the VerrazzanoProject custom resource to contain a list of corev1.Namespace resources (instead of a list of strings of namespace names)

Also fixed a problem in "make manifests" target because it was failing for me.

Fixes VZ-2289

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go [hunk 1]
```diff
@@ -4,13 +4,25 @@
...
+
+// ProjectTemplate contains the resources for a project
+type ProjectTemplate struct {
+	Namespaces []NamespaceTemplate `json:"namespaces"`
+}
+
+// VerrazzanoProjectSpec defines the desired state of VerrazzanoProject
 type VerrazzanoProjectSpec struct {
-	Namespaces []string `json:"namespaces"`
+	Template ProjectTemplate `json:"template"`
```
Comment (pfmackin) [inline application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go:25]: Why do we need ProjectTemplate? Usually templates are for some other resource that we will be creating.  I would expect to see NamespaceTemplate here.

File: application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go [hunk 2]
```diff
@@ -4,13 +4,25 @@
...
+// NamespaceTemplate has the metadata and spec of the underlying namespace
+type NamespaceTemplate struct {
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Metadata metav1.ObjectMeta    `json:"metadata"`
+	Spec     corev1.NamespaceSpec `json:"spec,omitempty"`
+}
+
+// ProjectTemplate contains the resources for a project
+type ProjectTemplate struct {
+	Namespaces []NamespaceTemplate `json:"namespaces"`
```
Comment (pfmackin) [inline application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go:20]: Shouldn't this be namespaceTemplates?  That is what the user will see in the CR and i think we agreed to use the term template

File: application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go (hunk 1 above)
Comment (mgianatagh) [reply application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go:25]: It is the "Template" section for the VerrazzanoProject resource

File: application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go (hunk 2 above)
Comment (mgianatagh) [reply application-operator/apis/clusters/v1alpha1/verrazzanoproject_types.go:20]: I don't think we want to rename this field.  Here is an example of the yaml that would be used:

`spec:
  template:
    namespaces:
      - metadata:
          name: multiclustertest
          labels:
            label1: "test1"
          annotations:
            annot1: "test1"
`

**Review by desagar — DISMISSED** [review]

**Review by desagar — APPROVED** [review]

**Pull request merged**